### PR TITLE
Standardize names; return exported types

### PR DESCRIPTION
**Pull request by nilsanderselde**
### Which issue this PR addresses:

Fixes [7970744](https://msazure.visualstudio.com/AzureRedHatOpenShift/_workitems/edit/7970744)

### What this PR does / why we "need" it:

Makes some names more consistent. Changes two functions so they return the exported interface type. Naming scheme for clients is `FooBars` for ARM clients, `foobarcli` for everything else.

#### Low priority

This PR can be the lowest priority to avoid inconveniencing other people. I will rebase and keep it up to date as needed.

### Test plan for issue:

Run existing tests

### Is there any documentation that needs to be updated for this PR?

No

### CODE REVIEW

**Comment by dkorzuno** [comment]
While consistent naming is important, the value of some of the proposed changes isn't that obvious to me. For example, I prefer "cli" to "somethingSomethingClient" when the name is local and it's obvious from the context which client it is. Struct fields, however, should be descriptive enough.

I let myself quote "The Go Programming Language" book here
> There is no limit on name length, but convention and style in Go programs lean toward short names, especially for local variables with small scopes; you are much more likely to see variables named i than theLoopIndex. Generally, the larger the scope of a name, the longer and more meaningful it should be.

As well as the [Effective Go](https://golang.org/doc/effective_go.html#names)
> Long names don't automatically make things more readable. A helpful doc comment can often be more valuable than an extra long name.

**Comment by nilsanderselde** [comment]
> While consistent naming is important, the value of some of the proposed changes isn't that obvious to me. For example, I prefer "cli" to "somethingSomethingClient" when the name is local and it's obvious from the context which client it is. Struct fields, however, should be descriptive enough.
> 
> I let myself quote "The Go Programming Language" book here
> 
> > There is no limit on name length, but convention and style in Go programs lean toward short names, especially for local variables with small scopes; you are much more likely to see variables named i than theLoopIndex. Generally, the larger the scope of a name, the longer and more meaningful it should be.
> 
> As well as the [Effective Go](https://golang.org/doc/effective_go.html#names)
> 
> > Long names don't automatically make things more readable. A helpful doc comment can often be more valuable than an extra long name.

The locally scoped variables don't need to be as descriptive, but they should be consistent. As far as the naming scheme, I don't really care what it is, as long as it's consistent. If everyone likes the `*cli` convention, then let's apply that everywhere. At the very least, though, I would argue it should be camelCase (`*Cli`).

**Comment by ihcsim** [comment]
I wonder if there is anything we can do to make sure everyone on the team understands what the coding convention and naming standards (if any) are, so we don't need to repeat this type of effort? Ideally, calling out misleading, ambiguous code before PRs are merged is more efficient than naming refactoring effort.

**Comment by mjudeikis** [comment]
Somehow when `Client` is appended all code looks very verbose. 
What is wrong with just having `groups`, `vm`, `kubernetes` ? 

@jim-minter  I think this needs discussions in some other forum than PR :)

**Comment by nilsanderselde** [comment]
> What is wrong with just having `groups`, `vm`, `kubernetes` ?

Nothing - but I think whatever naming scheme we decide on should be consistently applied.

**Comment by nilsanderselde** [comment]
I'll convert it to a draft so it's clear this is still a proposal

**Comment by nilsanderselde** [comment]
@mjudeikis I have removed the `*Client` suffix - can you take another look?

File: cmd/aro/operator.go [hunk 1]
```diff
@@ -55,63 +55,63 @@ func operator(ctx context.Context, log *logrus.Entry) error {
 		return err
 	}
 
-	kubernetescli, err := kubernetes.NewForConfig(restConfig)
+	k8s, err := kubernetes.NewForConfig(restConfig)
```
Comment (jim-minter) [inline cmd/aro/operator.go:58]:
```suggestion
	kubernetescli, err := kubernetes[client].NewForConfig(restconfig)
```
[client] optional

File: pkg/cluster/delete.go [hunk 2]
```diff
@@ -29,13 +29,13 @@ func (m *manager) deletePrivateDNSVirtualNetworkLinks(ctx context.Context, resou
 		return err
 	}
 
-	virtualnetworklinks, err := m.virtualnetworklinks.List(ctx, r.ResourceGroup, r.ResourceName, nil)
+	virtualNetworkLinks, err := m.virtualNetworkLinks.List(ctx, r.ResourceGroup, r.ResourceName, nil)
```
Comment (jim-minter) [inline pkg/cluster/delete.go:32]: +1

File: pkg/cluster/delete.go [hunk 3]
```diff
@@ -29,13 +29,13 @@ func (m *manager) deletePrivateDNSVirtualNetworkLinks(ctx context.Context, resou
...
 	if err != nil {
 		return err
 	}
 
-	for _, virtualnetworklink := range virtualnetworklinks {
-		err = m.virtualnetworklinks.DeleteAndWait(ctx, r.ResourceGroup, r.ResourceName, *virtualnetworklink.Name, "")
+	for _, virtualnetworklink := range virtualNetworkLinks {
```
Comment (jim-minter) [inline pkg/cluster/delete.go:37]:
```suggestion
	for _, virtualNetworkLink := range virtualNetworkLinks {
```
? ;-)

File: pkg/cluster/install.go [hunk 4]
```diff
@@ -134,45 +134,45 @@ func (m *manager) finishInstallation(ctx context.Context) error {
 	return err
 }
 
-// initializeKubernetesClients initializes clients which are used
+// initializeClients initializes clients which are used
```
Comment (jim-minter) [inline pkg/cluster/install.go:137]: tempted to leave this one be

**Review by jim-minter — COMMENTED** [review]
partial review - coming down to:

foos, err := x.virtualMachines.List() <- ARM/plural
foos, err := x.configcli.CoreV1().Pods().List() <- everything else/singular

File: cmd/aro/operator.go (hunk 1 above)
Comment (nilsanderselde) [reply cmd/aro/operator.go]: left as is per our discussion

File: pkg/backend/openshiftcluster/openshiftcluster.go [hunk 5]
```diff
@@ -44,14 +44,14 @@ type manager struct {
...
 	ocDynamicValidator validate.OpenShiftClusterDynamicValidator
 
-	groups         features.ResourceGroupsClient
+	resourceGroups features.ResourceGroupsClient
 	securityGroups network.SecurityGroupsClient
 
 	dns             dns.Manager
-	keyvault        keyvault.Manager
-	privateendpoint privateendpoint.Manager
+	keyVault        keyvault.Manager
```
[the remark this answers is not in the document]
Comment (nilsanderselde) [reply pkg/backend/openshiftcluster/openshiftcluster.go]: reverted

File: cmd/aro/rp.go [hunk 6]
```diff
@@ -32,7 +32,7 @@ import (
 )
 
 func rp(ctx context.Context, log *logrus.Entry) error {
-	_env, err := env.NewEnv(ctx, log)
```
[the remark this answers is not in the document]
Comment (nilsanderselde) [reply cmd/aro/rp.go]: reverted

File: hack/gendiscoverycache/gendiscoverycache.go [hunk 7]
```diff
@@ -27,22 +27,22 @@ import (
...
 
 func run(ctx context.Context, log *logrus.Entry) error {
-	kubeconfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+	k8sConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
 		clientcmd.NewDefaultClientConfigLoadingRules(),
 		&clientcmd.ConfigOverrides{},
 	)
 
-	restconfig, err := kubeconfig.ClientConfig()
+	restConfig, err := k8sConfig.ClientConfig()
```
[the remark this answers is not in the document]
Comment (nilsanderselde) [reply hack/gendiscoverycache/gendiscoverycache.go]: reverted, but didn't replace existing instances of `restConfig` with `restconfig`

File: hack/gendiscoverycache/gendiscoverycache.go [hunk 8]
```diff
@@ -27,22 +27,22 @@ import (
 const discoveryCacheDir = "pkg/util/dynamichelper/discovery/cache"
 
 func run(ctx context.Context, log *logrus.Entry) error {
-	kubeconfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+	k8sConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
```
[the remark this answers is not in the document]
Comment (nilsanderselde) [reply hack/gendiscoverycache/gendiscoverycache.go]: reverted

File: cmd/aro/operator.go [hunk 9]
```diff
@@ -55,63 +55,63 @@ func operator(ctx context.Context, log *logrus.Entry) error {
...
 	if err != nil {
 		return err
 	}
-	configcli, err := configclient.NewForConfig(restConfig)
+	config, err := configclient.NewForConfig(restConfig)
 	if err != nil {
 		return err
 	}
-	clustercli, err := maoclient.NewForConfig(restConfig)
+	cluster, err := maoclient.NewForConfig(restConfig)
```
[the remark this answers is not in the document]
Comment (nilsanderselde) [reply cmd/aro/operator.go]: maocli

File: pkg/backend/openshiftcluster/openshiftcluster.go [hunk 10]
```diff
@@ -44,7 +44,7 @@ type manager struct {
 
 	ocDynamicValidator validate.OpenShiftClusterDynamicValidator
 
-	groups         features.ResourceGroupsClient
+	resourceGroups features.ResourceGroupsClient
```
Comment (jim-minter) [inline pkg/backend/openshiftcluster/openshiftcluster.go:47]: rofl, evidently these fields aren't even used any more!  I'll open a follow-up to sort that out.

**Review by jim-minter — APPROVED** [review]
LGTM.  Thanks for doing this @nilsanderselde.  This PR is a smaller step than it was before, but it definitely takes us forwards.  Will merge on green.

**Pull request merged**